src/app/components/pages/MyProgress.tsx

```diff
@@ -37,8 +38,44 @@ const statistics = {
   tagColWidth: "col-lg-12",
 };
 
-interface MyProgressProps extends RouteComponentProps<{ userIdOfInterest: string }> {
+const QuestionParts = ({ progress }: { progress: MyProgressState | undefined }) => {
+  return (
+    <div className="mt-4">
+      <h4>Question parts correct by Type</h4>
+      <Row>
+        {statistics.questionTypeStatsList.map((qType: string) => {
+          const correct = progress?.correctByType?.[qType] ?? null;
+          const attempts = progress?.attemptsByType?.[qType] ?? null;
+          const percentage = safePercentage(correct, attempts);
+          return (
+            <Col key={qType} className={`${statistics.typeColWidth} mt-2 type-progress-bar`}>
+              <div className={"px-2"}>{HUMAN_QUESTION_TYPES[qType]} questions correct</div>
+              <div className={"px-2"}>
+                <ProgressBar percentage={percentage ?? 0}>
+                  {percentage == null ? "No data" : `${correct} of ${attempts}`}
+                </ProgressBar>
+              </div>
+            </Col>
+          );
+        })}
+      </Row>
+    </div>
+  );
+};
+
+const canCurrentUserViewOtherUserData = ({
+  viewingOwnData,
+  user,
+}: {
+  viewingOwnData: boolean;
   user: PotentialUser;
+}) => {
+  if (!viewingOwnData && isTeacherOrAbove(user)) return true;
+  else if (!viewingOwnData && !isTeacherOrAbove(user)) return false;
+};
+
+interface MyProgressProps extends RouteComponentProps<{ userIdOfInterest: string }> {
+  user: PotentialUser | LoggedInUser;
 }
```

Comment:
This suggestion is so small that it's barely worth mentioning, but It would be nice to add a blank line here to separate the interface and the constant.